workbench-core/audit/src/auditMiddleware.ts

```diff
@@ -49,10 +49,12 @@ export function WithAudit(
     }
     const extractor = auditConfig.extractor ?? BaseExtractor;
     const metadata: Metadata = extractor.getMetadata(req, res);
-    next();
-    metadata.statusCode = res.statusCode;
+    // TODO: Correctly propogate statusCode. StatusCode is not correct when an error is thrown. The error handler is executed after audit log is written, therefore status code is inaccurate
+    // next();
+    // metadata.statusCode = res.statusCode;
```

Comment:
@htztomic Are you ok with me removing this field? I've found that if an error is thrown, the HTTP status code in the logs are not accurate. The actual status code is returned [here](https://github.com/aws-solutions/solution-spark-on-aws/blob/523d92babeea50db8219e2a7a5660de9c58dd287/solutions/swb-app/src/errorHandlers.ts#L22), but it's executed after the logs are written. 

I think this is ok, because customers can still get the correct status code by checking API GW access logs. 

Example API GW access log is shown here
```
```
{
    "stage": "dev",
    "requestTime": "14/Dec/2022:22:36:39 +0000",
    "requestId": "26117525-794f-419b-9f61-793c19b266a8",
    "integrationRequestId": "37c8b6ea-f4a6-4668-b68e-83d3fff067ae",
    "status": "400",
    "apiId": "5diaqbnwy2",
    "resourcePath": "/{proxy+}",
    "path": "/dev/costCenters/",
    "resourceId": "zu1kdw",
    "httpMethod": "POST",
    "sourceIp": "<redacted>",
    "userAgent": "PostmanRuntime/7.29.2"
}
```

Reply:
Thanks @nguyen102 . will approve and drop a ticket on the backlog to revisit.

Reply:
@nguyen102 , should we configure logger to be executed the last?

Reply:
I spent a couple of hours trying to get the middleware to be executed last, but I wasn't able to get it working. This would need to be investigated further in MAFoundation-545

Reply:
I vaguely remember a discussion on firing the audit logging last at the end of the `next()` calls. I think there was an issue with express where execution terminates on `res.send()`. This is definitely something we can revisit in MAFoundation-545 just to ensure I'm remembering correctly and we didn't miss anything.